src/gui/sharedialog.cpp

```diff
@@ -558,6 +572,46 @@ void ShareDialog::slotPushButtonCopyLinkPressed()
     clipboard->setText(_shareUrl);
 }
 
+void ShareDialog::slotCheckBoxEditingClicked()
+{
+    ShareDialog::setPublicUpload(_ui->checkBox_editing->checkState() == Qt::Checked);
+}
+
+void ShareDialog::setPublicUpload(bool publicUpload)
+{
+    _ui->checkBox_editing->setEnabled(false);
+    _pi_editing->startAnimation();
+
+    QUrl url;
+    QList<QPair<QString, QString> > requestParams;
+    QByteArray verb("PUT");
+
+    QString value = QString::fromLatin1(publicUpload ? "true" : "false");
+
+    url = Account::concatUrlPath(_account->url(), QString("ocs/v1.php/apps/files_sharing/api/v1/shares/%1").arg(_public_share_id));
```

Comment:
mmm chaning to QLating1String fails because it is a QList<QPair<QString, QString> >
Maybe we can fix that in a later PR... Becuase it needs changes at more places. Probably it could become a QList<QPair<QLatin1String, QString> >

You are right about the const stuff.